packages/x-data-grid/src/hooks/features/rows/useGridRowsMeta.ts

```diff
@@ -73,55 +70,55 @@ export const useGridRowsMeta = (
     () => gridDimensionsSelector(apiRef.current.state).rowHeight,
   );
 
-  const hydrateRowsMeta = React.useCallback(() => {
-    hasRowWithAutoHeight.current = false;
-
-    const calculateRowProcessedSizes = (row: GridRowEntry) => {
-      if (!rowsHeightLookup.current[row.id]) {
-        rowsHeightLookup.current[row.id] = {
-          sizes: { baseCenter: rowHeight },
-          isResized: false,
-          autoHeight: false,
-          needsFirstMeasurement: true, // Assume all rows will need to be measured by default
-        };
-      }
+  const getRowHeightEntry: GridRowsMetaPrivateApi['getRowHeightEntry'] = (rowId) => {
+    let entry = heightCache.get(rowId);
+    if (entry === undefined) {
+      entry = {
+        content: rowHeight,
+        spacingTop: 0,
+        spacingBottom: 0,
+        detail: 0,
+        autoHeight: false,
+        needsFirstMeasurement: true,
+      };
+      heightCache.set(rowId, entry);
+    }
+    return entry;
+  };
 
-      const { isResized, needsFirstMeasurement, sizes } = rowsHeightLookup.current[row.id];
-      let baseRowHeight = typeof rowHeight === 'number' && rowHeight > 0 ? rowHeight : 52;
-      const existingBaseRowHeight = sizes.baseCenter;
+  const processHeightEntry = React.useCallback(
+    (row: GridRowEntry) => {
+      // HACK: rowHeight trails behind the most up-to-date value just enough to
+      // mess the initial rowsMeta hydration :/
+      const baseRowHeight = gridDimensionsSelector(apiRef.current.state).rowHeight;
```

Comment:
@cherniavskii The recent change you made to pull `rowHeight` from `useGridSelector` coupled with the changes of this PR created this issue :| It's a symptom of [this](https://www.notion.so/mui-org/mui-x-Data-Grid-refactoring-120cbfe7b6608044bedad0bbc8e87a48?pvs=4#121cbfe7b6608089ba62c43dfd69f4c3).